lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -443,146 +517,243 @@ def create_autoscaling_group(connection, module):
     health_check_type = module.params.get('health_check_type')
     default_cooldown = module.params.get('default_cooldown')
     wait_for_instances = module.params.get('wait_for_instances')
-    as_groups = connection.get_all_groups(names=[group_name])
+    as_groups = connection.describe_auto_scaling_groups(AutoScalingGroupNames=[group_name])
```

Comment:
I'm fine with this being a followup PR. Error handling needs to be added, but that'd increase the diffstat of this PR pretty significantly, and since the handling wasn't there before I wouldn't block the merge over it.